test/e2e/storage/testsuites/provisioning.go

```diff
@@ -527,6 +528,271 @@ func (p *provisioningTestSuite) DefineTests(driver storageframework.TestDriver,
+		biggerROClaim.Name = "bigger-claim"
+		biggerROClaim.Spec.Resources = v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceStorage: biggerSize,
+			},
+		}
+		biggerROClaim.Spec.AccessModes = []v1.PersistentVolumeAccessMode{
+			v1.PersistentVolumeAccessMode(v1.ReadOnlyMany),
+		}
+
+		// Create claim w/ same size as original
+		sameSizeROClaim := l.testCase.Claim.DeepCopy()
+		sameSizeROClaim.Name = "same-size-claim"
+		sameSizeROClaim.Spec.AccessModes = []v1.PersistentVolumeAccessMode{
+			v1.PersistentVolumeAccessMode(v1.ReadOnlyMany),
+		}
+
+		// For use in TestDynamicProvisioning - test expected content exists in
+		// restored volume
+		l.testCase.PvCheck = func(ctx context.Context, claim *v1.PersistentVolumeClaim) {
+			ginkgo.By("checking whether the created volume has the pre-populated data")
+			tests := []e2evolume.Test{
+				{
+					Volume:          *storageutils.CreateVolumeSource(claim.Name, false /* readOnly */),
+					Mode:            pattern.VolMode,
+					File:            "index.html",
+					ExpectedContent: expectedContent,
+				},
+			}
+			e2evolume.TestVolumeClientSlow(ctx, f, testConfig, nil, "", tests)
+		}
+
+		for _, claim := range []*v1.PersistentVolumeClaim{sameSizeROClaim, biggerROClaim} {
+			l.testCase.Claim = claim
+			l.testCase.TestDynamicProvisioning(ctx)
+
+			// Do not proceed to fs size check for block file systems, or drivers
+			// that support fs resize in ROX mode
+			if pattern.VolMode == "Block" {
+				continue
+			}
+
+			if dInfo.Capabilities[storageframework.CapFSROXResizeFromSourceNotSupported] {
+				continue
+			}
+
+			// Create pod w/ claim so we can get the mount path of fs
+			claim, err := l.testCase.Client.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+			framework.ExpectNoError(err)
+			createdClaims := []*v1.PersistentVolumeClaim{claim}
+			var podConfig *e2epod.Config = &e2epod.Config{
+				NS:            f.Namespace.Name,
+				PVCs:          createdClaims,
+				NodeSelection: testConfig.ClientNodeSelection,
+				PVCsReadOnly:  true,
+			}
+			pod, err := e2epod.CreateSecPod(ctx, l.testCase.Client, podConfig, l.testCase.Timeouts.DataSourceProvision)
+
+			framework.ExpectNoError(err, "Failed to create pod: %v", err)
+			mountpath := findVolumeMountPath(pod, claim)
+			gomega.Expect(mountpath).ShouldNot(gomega.BeEmpty())
+
+			// Save filesystem size of the restored volume.
+			restoredFSSize, err := getFilesystemSizeBytes(pod, mountpath)
+			framework.ExpectNoError(err, "Failed to obtain filesystem size of a volume mount: %v", err)
+
+			msg := fmt.Sprintf("Filesystem resize occurred even though it is readonly. "+
+				"Restored fs size: %v bytes is not the same as origin fs size: %v bytes.\n"+
+				"HINT: Your driver needs to check the volume and skip resize if it's readonly.",
+				restoredFSSize, originFSSize)
+			gomega.Expect(restoredFSSize).Should(gomega.BeEquivalentTo(originFSSize), msg)
```

Comment:
In the case where the sameSize claim is created, the restored size should be the same.

In the case where the biggerSize claim is created, the restored size should be bigger.

Reply:
The whole point of this test is to test that in ROX mode, no matter what size you request in the PVC, the cloned size will be the same as the original, since a resize cannot occur (bc the fs is readonly). Are we not aligned on this? See https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/1088 and https://github.com/kubernetes/kubernetes/pull/114325 for context. I also have an internal test plan I'll send along.

It's a little hard to follow because we have both l.testCase.Claim and l.pvc, but you can see [here](https://github.com/kubernetes/kubernetes/blob/dd985dcdcde8d9f2b89caec382fdf9db29b845dc/test/e2e/storage/testsuites/provisioning.go#L172) they are the same underlying object. So then, by setting:

```
l.pvc.Spec.DataSourceRef = dataSourceRef
```
above, then we create both sameSizeROClaim and biggerROClaim as deep copies of that claim:

```
biggerROClaim := l.testCase.Claim.DeepCopy()
...
sameSizeROClaim := l.testCase.Claim.DeepCopy()
```
So both of those are being restored from the same cloned data source, and even though we request a larger size in the biggerPVC, since we're in ROX mode, it should restore to the original size since it can't do a resize.

Reply:
As a result of other comment I removed file size checking here so this point is now moot, so resolving.